lib/js_defer.js

```diff
@@ -1266,10 +1278,26 @@ deferJsNs.DeferJs.prototype.registerScriptTags = function() {
   this.state_ = deferJsNs.DeferJs.STATES.SCRIPTS_REGISTERED;
   var scripts = document.getElementsByTagName('script');
   var len = scripts.length;
+  var syncScript = 0;
+  var deferScript = 0;
   for (var i = 0; i < len; ++i) {
     var script = scripts[i];
     if (script.getAttribute('type') == this.psaScriptType_) {
-      this.addNode(script);
+      // async attribute is priority over defer attribute
+      if (script.getAttribute('async') !== null) {
+        this.addNode(script);
+      }
+      // sync scripts are placed before defer scripts and defer scripts are placed before async scripts
+      else if (script.getAttribute('defer') !== null) {
+        // add script with defer attribute after sync script
+        this.addNode(script, syncScript + deferScript);
+        deferScript++;
+      }
+      else {
+        // add before deferjs script
+        this.addNode(script, syncScript);
+        syncScript++;
```

Comment:
If the code declare (in this order) a defer then a sync scripts, they will both be at position 0. Is that what we whant ?

Reply:
The logic should be to insert the sync script before the defer script in the queue. Thus, inserting the sync script at the position 0, will move the defered script at the position 1.
I think that the implementation is correct. I will try the case to verify.

Reply:
In this case, when the sync script will be set at the position 0, the deferred script will be set at the position 1.
It's tested with the test using the defer_order.html file.